lib/migration.js

```diff
@@ -313,32 +335,53 @@ function mixinMigration(PostgreSQL) {
   PostgreSQL.prototype.createTable = function(model, cb) {
     const self = this;
     const name = self.tableEscaped(model);
+    const modelDef = this.getModelDefinition(model);
+
+    // collects all extenstions need to be created
+    let createExtensions;
+    Object.keys(this.getModelDefinition(model).properties).forEach(function(propName) {
+      const prop = modelDef.properties[propName];
+
+      // checks if dataType is set to uuid
+      const postgSettings = prop.postgresql;
+      if (postgSettings && postgSettings.dataType && postgSettings.dataType === 'UUID'
+         && postgSettings.defaultFn && postgSettings.extension) {
+        createExtensions += 'CREATE EXTENSION IF NOT EXISTS "' + postgSettings.extension + '";';
+      }
+    });
+    // default extension
+    if (!createExtensions) {
+      createExtensions = 'CREATE EXTENSION IF NOT EXISTS "uuid-ossp";';
+    }
 
     // Please note IF NOT EXISTS is introduced in postgresql v9.3
-    self.execute('CREATE SCHEMA ' +
+    self.execute(
+      createExtensions +
+      'CREATE SCHEMA ' +
       self.escapeName(self.schema(model)),
-    function(err) {
-      if (err && err.code !== '42P06') {
-        return cb && cb(err);
-      }
-      self.execute('CREATE TABLE ' + name + ' (\n  ' +
-          self.propertiesSQL(model) + '\n)',
-      function(err, info) {
-        if (err) {
-          return cb(err, info);
+      function(err) {
+        if (err && err.code !== '42P06') {
+          return cb && cb(err);
```

Comment:
Why return `cb` and not just `cb(err)`?

Reply:
I didn't change this part.  ( review it with whitespace hided

Reply:
Oh sorry I refreshed, sounds good 👍